### DISCUSSION TITLE
Target ES Module Syntax for Cloudflare Workers

### DISCUSSION THREAD

**Discussion by geelen** [nuxt/nuxt#18246] [Ideas]
At the moment, the Nuxt 3 build targets serviceworker syntax for CF Workers, but the newer syntax using Modules has some better performance and will be the preferred API going forward: https://developers.cloudflare.com/workers/runtime-apis/fetch-event#syntax-module-worker

Key differences are:
* no `addEventListener`, instead `export default { fetch() { ... }}`
* no globals, instead the `env` argument to `fetch` provides these, and you're required to pass them to the modules that require them e.g. `env.__STATIC_CONTENT`

Having a module-syntax compatible output for a Nuxt build will also make it trivial to integrate with the upcoming release of Cloudflare Pages which will be bringing Workers integration.

Happy to be available to help make this happen in time for launch!

**Comment by pi0** [comment]
It is already in todo list: https://github.com/nuxt/framework/issues/484
